File: rust/hlsocial/src/api/v1.rs

use sqlx::FromRow;
use chrono::NaiveDate;
use serde::{Serialize, Deserialize};

pub mod token;
pub mod login;
pub mod user;
pub mod middleware;

use super::AppState;


#[derive(Serialize, FromRow, Debug)]
pub struct User {
    user_id: i64,
    first_name: Option<String>,
    second_name: Option<String>,
    email: String,
    gender: Option<String>,
    birthdate: Option<NaiveDate>,
    biography: Option<String>,
    city: Option<String>,
}

#[derive(Deserialize, FromRow, Debug)]
pub struct UserPwhash {
    user_id: i64,
    pwhash: String,
}

#[derive(Deserialize, FromRow, Debug)]
pub struct UserCreate {
    first_name: Option<String>,
    second_name: Option<String>,
    email: String,
    passwd: String,
    gender: Option<String>,
    birthdate: Option<NaiveDate>,
    biography: Option<String>,
    city: Option<String>,
}

#[derive(Deserialize, FromRow, Debug)]
pub struct UserUpdate {
    user_id: i64,
    first_name: Option<String>,
    second_name: Option<String>,
    email: String,
    gender: Option<String>,
    birthdate: Option<NaiveDate>,
    biography: Option<String>,
    city: Option<String>,
}

#[derive(Deserialize, FromRow, Debug)]
pub struct UserCreateResult {
    user_id: i64,
}

File: wi/hlsocial/src/api/v1.ts


export interface User {
  user_id?: number,
  first_name?: string | null,
  second_name?: string | null,
  email?: string,
  gender?: string | null,
  birthdate?: number | null,
  biography?: string | null,
  city?: string | null,
}

export interface UserLoginByEmail {
  email?: string,
  passwd?: string,
}

export interface UserPwhash {
  user_id?: number,
  pwhash?: String,
}

export interface UserCreate {
  first_name?: string | null,
  second_name?: string | null,
  email?: string,
  passwd?: string,
  gender?: string | null,
  birthdate?: number | null,
  biography?: string | null,
  city?: string | null,
}

export interface UserUpdate {
  user_id?: number,
  first_name?: string | null,
  second_name?: string | null,
  email?: string,
  gender?: string | null,
  birthdate?: number | null,
  biography?: string | null,
  city?: string | null,
}

export interface UserCreateResult {
  user_id?: number,
}

export interface UserLoginResult {
  user_id?: number,
}
